app/Http/Controllers/DashboardController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\RoomResult;
+use App\Models\RoomOpenEvent;
+use Inertia\Inertia;
+use Inertia\Response;
+use Carbon\CarbonImmutable;
+use Carbon\CarbonPeriod;
+
+class DashboardController extends Controller
+{
+	public function index(): Response
+	{
+		$user = auth()->user();
+
+		$rooms = $user->rooms()->pluck("id", "name");
+		$roomIds = $rooms->values();
+		$roomsOpenedCount = RoomOpenEvent::whereIn(
+			"room_id",
+			$roomIds
+		)->count();
+
+		$now = CarbonImmutable::now();
+		$weekStartDate = $now->startOfWeek()->format("Y-m-d");
+		$weekEndDate = $now->endOfWeek()->format("Y-m-d");
+		$period = CarbonPeriod::create($weekStartDate, "1 day", $weekEndDate);
+
+		$roomsOpenedThisWeek = [];
+
+		foreach ($period as $key => $date) {
+			array_push(
+				$roomsOpenedThisWeek,
+				RoomOpenEvent::whereIn("room_id", $roomIds)
+					->where(
+						"created_at",
+						">=",
+						$date->setTime(0, 0, 0)->toDateTimeString()
+					)
+					->where(
+						"created_at",
+						"<=",
+						$date->setTime(23, 59, 59)->toDateTimeString()
+					)
+					->count()
+			);
+		}
```

Comment:
I would say it would be more readable to just use collection instead of array and then use method `->push()`